File: keytocrypto.py
from sys import stdin,stdout

ciphertext,secret = stdin.readline().strip(),stdin.readline().strip()
ans = []
for i in range(min(len(secret),len(ciphertext))):
    ans.append(chr((ord(ciphertext[i]) - ord(secret[i]) + 26) % 26 + ord('A')))
for i in range(len(secret), len(ciphertext)):
    ans.append(chr((ord(ciphertext[i]) - ord(ans[i - len(secret)])) % 26 + ord('A')))
stdout.write("".join(char for char in ans))

File: keytocrypto.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    string ciphertext, secret; cin >> ciphertext >> secret;
    vector<char> ans;
    for (int i = 0; i < min(secret.length(),ciphertext.length()); i++) {
        ans.push_back((ciphertext[i] - secret[i] + 26) % 26 + 'A');
    }
    for (int i = secret.length(); i < ciphertext.length(); i++) {
        ans.push_back((ciphertext[i] - ans[i - secret.length()] + 26) % 26 + 'A');
    }
    for (char c: ans) cout << c;
    return 0;
}